### DISCUSSION TITLE
http notifications with ntfy

### DISCUSSION THREAD

**Discussion by AbsoluteXYZero** [containrrr/watchtower#2035] [Q&A]
Hello all, as the Tittle suggests I'm running watchtower with docker and would like to receive notifications via ntfy (also on docker) whenever a container gets updated however I'm behind a CGNAT and my server doesn't have an https domain so I'm just trying to setup notifications through my local network.

I get errors in watchtower when I try to use my internal IP (ntfy://192.168.x.x....) for the URL in docker compose because it's not an https (and yes I have tried putting the leading https in but that just makes for addition errors.

I've read the documentation and see that there is a Props option as ?key=value.... And the default Scheme is https... But I'm not sure how to make use of this info correctly...

I've tried the following but it doesn't appear to be in the correct syntax.
Watchtower_notification_url: ntfy://192.168.x.x:7200/?title=Updates/?Scheme=http

Help me out here please... Any ideas? Are local notifications possible at all this way?

Thanks in advance

![Screenshot_20241003_163640_Firefox](https://github.com/user-attachments/assets/7d232945-03fb-4119-aec3-0e909157f009)
![Screenshot_20241003_163836_Firefox](https://github.com/user-attachments/assets/eb7fb402-3b5e-4bc0-8e55-e699fa5006b9)

**Comment by AbsoluteXYZero** [comment]
Solved the issue myself. Gonna keep the post up to help others. Needed to change my docker compose text to the following for it to work:

WATCHTOWER_NOTIFICATIONS: shoutrrr
WATCHTOWER_NOTIFICATION_URL: ntfy://192.168.0.137:7200/Updates?scheme=http

**Reply by CoolnsX** [reply]
A big thanks to you brother

**Reply by eleventhaus** [reply]
TY

I'll add that [if you want to include authentication](https://containrrr.dev/shoutrrr/v0.8/services/overview/), the url would look like this:
ntfy://{username}:{password}@ntfy.sh/{topic}?scheme=http

**Reply by CoolnsX** [reply]
for token based authentication

WATCHTOWER_NOTIFICATIONS=shoutrrr
WATCHTOWER_NOTIFICATION_URL=ntfy://:token@domain.tld/topic?icon=icon_url

**Comment by rjhoppe** [comment]
For anyone else attempting to run this using portainer and nginx with a public domain this was what worked for me:

- WATCHTOWER_NOTIFICATIONS=shoutrrr
- WATCHTOWER_NOTIFICATION_URL=ntfy://subdomain.domain/[your-topic-here]?scheme=https

Both my ntfy and watchtower services are on the same VPS
